### PR TITLE
Migrate `rustc_attr` crate diagnostics

### PR DESCRIPTION
**Pull request by hampuslidin**
Hi!

This is my first PR to the rustc project, excited to be part of the development! This PR is part of the diagnostics effort, to make diagnostics translatable.

@rustbot label +A-translation

### CODE REVIEW

**Comment by hampuslidin** [comment]
I think this PR is ready for review. There are a couple of FIXMEs in there I’d like some feedback on.

**Comment by hampuslidin** [comment]
r? rust-lang/diagnostics

File: compiler/rustc_attr/src/session_diagnostics.rs [hunk 1]
```diff
@@ -0,0 +1,397 @@
...
+}
+
+#[derive(SessionDiagnostic)]
+#[diag(attr::missing_issue, code = "E0547")]
+pub(crate) struct MissingIssue {
+    #[primary_span]
+    pub span: Span,
+}
+
+// FIXME: This diagnostic is identical to `IncorrectMetaItem`, barring the error code. Consider
```
Comment (davidtwco) [inline compiler/rustc_attr/src/session_diagnostics.rs:166]: I've suggested on https://github.com/rust-lang/rust/pull/100831#discussion_r951403293 supporting `SessionDiagnostic` enums, which is how I think I'd prefer to deduplicate this. If we move from error codes to slugs in future, which I'd like to, then it goes away as a problem somewhat too. This is fine for now.

File: compiler/rustc_attr/src/session_diagnostics.rs [hunk 2]
```diff
@@ -0,0 +1,397 @@
...
+// changing this to `IncorrectMetaItem`. See #51489.
+#[derive(SessionDiagnostic)]
+#[diag(attr::incorrect_meta_item, code = "E0551")]
+pub(crate) struct IncorrectMetaItem2 {
+    #[primary_span]
+    pub span: Span,
+}
+
+// FIXME: Why is this the same error code as `InvalidReprHintNoParen` and `InvalidReprHintNoValue`?
```
Comment (davidtwco) [inline compiler/rustc_attr/src/session_diagnostics.rs:175]: If the error code assignments seem strange then that's a pre-existing issue that we can leave for now, error codes are due a once over anyway.

**Review by davidtwco — APPROVED** [review]
LGTM, thanks!

**Comment by davidtwco** [comment]
r=me after rebase

**Comment by hampuslidin** [comment]
Rebased!

**Pull request merged**